tpr/kind.go

```diff
@@ -0,0 +1,40 @@
+package tpr
+
+import (
+	"strings"
+
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+/*
+	Code based on:
+	https://github.com/kubernetes/kubernetes/blob/6a4d5cd7cc58e28c20ca133dab7b0e9e56192fe3/pkg/registry/extensions/thirdpartyresourcedata/util.go
+*/
+
+// extractKindAndGroup extracts kind and group from a name. For details see
+// Config.Name godoc.
+func extractKindAndGroup(name string) (kind string, group string, err error) {
+	parts := strings.Split(name, ".")
+	if len(parts) < 3 {
+		return "", "", microerror.MaskAnyf(unexpectedlyShortResourceNameError, "%s, expected at least <kind>.<domain>.<tld>", name)
```

Comment:
We've been using a different variable name in [aws-operator](https://github.com/giantswarm/aws-operator/blob/320bda04e31d246a172d6af81fb6656dd93807b5/service/create/load_balancer.go#L108), which I find reads a little bit better, and is more generic.